### PR TITLE
KFSPTS-21715 add Security Request wizard

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Review by cah292 — CHANGES_REQUESTED** [review]
I found a few minor items that would be good to clean up. Other than that, the changes look good.

File: src/main/java/edu/cornell/kfs/ksr/service/impl/SecurityRequestDocumentServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+public class SecurityRequestDocumentServiceImpl implements SecurityRequestDocumentService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(SecurityRequestDocumentServiceImpl.class);
+
+    private BusinessObjectService businessObjectService;
+
+    @Override
+    public List<SecurityGroup> getActiveSecurityGroups() {
+        Map<String, Object> hashMap = new HashMap<String, Object>();
+        hashMap.put("active", true);
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/ksr/service/impl/SecurityRequestDocumentServiceImpl.java:21]: updated

File: src/main/java/edu/cornell/kfs/ksr/web/struts/SecurityRequestWizardForm.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    public SecurityRequestWizardForm() {
+        super();
+    }
+
+    /**
+     * Overriding to allow the process wizard call (since the wizard form is not
+     * stored in session)
+     * 
+     * @see org.kuali.rice.kns.web.struts.form.KualiForm#shouldMethodToCallParameterBeUsed(java.lang.String,
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/ksr/web/struts/SecurityRequestWizardForm.java:29]: updated

File: src/main/webapp/WEB-INF/institutional-struts-config.xml [hunk 3]
```diff
@@ -121,11 +123,11 @@
...
-      type="edu.cornell.kfs.fp.document.web.struts.CuDistributionOfIncomeAndExpenseAction" scope="request"
-      parameter="methodToCall" validate="true" attribute="KualiForm">
-      <set-property property="cancellable" value="true" />
-      <forward name="basic" path="/jsp/fp/DistributionOfIncomeAndExpense.jsp" />
-    </action> 
+      		type="edu.cornell.kfs.fp.document.web.struts.CuDistributionOfIncomeAndExpenseAction" scope="request"
+      		parameter="methodToCall" validate="true" attribute="KualiForm">
+      		<set-property property="cancellable" value="true" />
+     		<forward name="basic" path="/jsp/fp/DistributionOfIncomeAndExpense.jsp" />
+    	</action>
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/webapp/WEB-INF/institutional-struts-config.xml:130]: This formatting was off so I fixed it. I will undo the change.

File: src/main/java/edu/cornell/kfs/ksr/web/struts/SecurityRequestWizardForm.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public SecurityGroup getSecurityGroup() {
+        return securityGroup;
+    }
+
+    public void setSecurityGroup(SecurityGroup securityGroup) {
+        this.securityGroup = securityGroup;
+    }
+
+	public void setSecurityGroups(List<SecurityGroup> securityGroups) {
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/ksr/web/struts/SecurityRequestWizardForm.java]: updated

**Comment by db483** [comment]
I added the requested changes and this is ready for review again.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass, and the SQL changes are also ready. One quick note: KFS already has its own constants for the "active" property name, but I'm okay with moving these changes forward as-is and we can clean up the "active" property constants in a future user story.

Merging.

**Pull request merged**